src/cmap/message_stream.ts

```diff
@@ -165,12 +171,20 @@ function processIncomingData(stream: MessageStream, callback: Callback<Buffer>)
   let ResponseType = messageHeader.opCode === OP_MSG ? BinMsg : Response;
   if (messageHeader.opCode !== OP_COMPRESSED) {
     const messageBody = message.slice(MESSAGE_HEADER_SIZE);
-    stream.emit('message', new ResponseType(message, messageHeader, messageBody));
 
-    if (buffer.length >= 4) {
+    // If we are a monitoring message stream using the streaming protocol and
+    // there is more in the buffer that can be read, skip processing since we
+    // want the last hello command response that is in the buffer.
+    if (stream.isStreamingProtocol && buffer.length >= 4) {
```

Comment:
I did it like this so that if there is at least 4 bytes left in the buffer, that will be the 4 bytes that can be read for the size of the next message. If those 4 bytes exist, we know there is another whether it is entirely on the buffer or not and the code will recurse until it gets it. This way we avoid potentially reading more messages when there are partials and attempt to truly get the very last message that was sent.

Reply:
We don't want to get the the last message that was _sent_ (that is unknowable, actually).  We want to get the last message that was _received_.   So the check should be, semantically, whether the buffer contains another full message.  Only then should we skip the current message. This check seems straightforward to implement.  The logic 
 is basically already implemented earlier in the method.  It just needs to be extracted into a function that, perhaps, could be re-used.

Reply:
So basically extract a method like `readLastMessage()` that checks the length of the next message in the buffer, checks if more than that number of bytes are present, then recursively checks the same again until a partial is there, and take the one before it?

Reply:
That would work (though it doesn't have to be implemented recursively).

Reply:
Updated the check to peek the size and added extra test for the case where a partial message is present.